File: Abyss/Timer.cs
using System;

namespace Capthed.Abyss {
	
	public static class Timer {

		private static long delta;
		// Returns the time from the last tick to this one.
		public static long Delta { get { return delta; }}
		public static readonly int SECOND = 1000;
		private static readonly DateTime Jan1st1970 = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);

		// Returns the current time in ms.
		public static long GetTime() {
			return (long)(DateTime.UtcNow - Jan1st1970).TotalMilliseconds;;
		}

		public static void SetDelta(long ms) {
			delta = ms;
		}
	}
}

File: src/com/capthed/abyss/Timer.java
package com.capthed.abyss;

public abstract class Timer {

	public static final float SECOND = 1000.0f;
	private static long startTime = 0;
	private static long delta;
	
	public static void start() {
		startTime = getTime();
	}
	
	/** @return The current time in ms */
	public static long getTime() { return System.currentTimeMillis(); }
	
	/** @return The time passed from last frame to this one in seconds*/
	public static double getDelta() { 
		double temp = delta / (double) SECOND;
		return temp; 
	}
	
	static void setDelta(long d) { delta = d; }

	/** @return The time passed since the program has been started in seconds */
	public static double getTimeRunning() {
		return (getTime() - startTime) / (double) SECOND;
	}
}